app/queries/admin/moderators_query.rb

```diff
@@ -4,29 +4,29 @@ class ModeratorsQuery
       state: :trusted
     }.with_indifferent_access.freeze
 
+    NAMES_FOR_POTENTIAL = %i[banned warned trusted comment_banned].freeze
+
     def self.call(relation: User.all, options: {})
       options = DEFAULT_OPTIONS.merge(options)
       state, search = options.values_at(:state, :search)
 
       relation = if state.to_s == "potential"
                    relation.where(
-                     "id NOT IN (SELECT user_id FROM users_roles WHERE role_id = ?)
-                     AND id NOT IN (SELECT user_id FROM users_roles WHERE role_id = ?)",
-                     role_id_for(:trusted),
-                     role_id_for(:banned),
+                     "id NOT IN (SELECT user_id FROM users_roles WHERE role_id IN (?))",
+                     potential_role_ids,
                    ).order("users.comments_count" => :desc)
                  else
                    relation.joins(:roles)
-                     .where(users_roles: { role_id: role_id_for(state) })
+                     .where(users_roles: { role_id: Role.find_by(name: state)&.id })
                  end
 
       relation = search_relation(relation, search) if search.presence
 
       relation
     end
 
-    def self.role_id_for(role)
-      Role.find_by(name: role)&.id
+    def self.potential_role_ids
+      Role.where(name: NAMES_FOR_POTENTIAL).select(:id)
```

Comment:
Feel like this could be refactored, cached, or frozen somehow...

Reply:
Rails has a built-in `ids` method for this purpose, which is even simpler than `pluck(:id)` (though it creates the exact same query):

```suggestion
    def self.potential_role_ids
      Role.where(name: NAMES_FOR_POTENTIAL).ids
```